docs/DeveloperGuide.md

```diff
@@ -1,11 +1,461 @@
+Command executed by user `project /create -title <title> -desc <description> -dur <duration> -sd <sprint interval>`
+is passed, the following operations are implemented:
+    * UI receives user input and passes it to Parser class.
+    * Parser checks if input format is valid, and executes a corresponding AddProjectCommand object.
+    * A new project is created, and added to project manager.
+
+#### Select Project
+### Task
+#### Add Task
+A task is created following the creation of a project, with a clear title, description 
+and priority of the task.
+Command executed by user `task /add -title <title> -desc <description> -priority <priority>`
+is passed, the following operations are implemented:
+    * UI receives user input and passes it to Parser class.
+    * Parser checks if the input format is valid and if a project exists, and executes a corresponding AddTaskCommand 
+    object.
+    * A new task is created, and added to project manager.
+    
+#### View Task
+#### Delete Task
+An existing task is deleted when the user wishes to remove a completed or unnecessary task. The task
+ID is provided.
+Command executed by user `task /del -id`
+is passed, the following operations are implemented:
+    * UI receives user input and passes it to Parser class.
+    * Parser checks if the id entered is valid, and executes a corresponding DeleteTaskCommand 
+    object.
+    * The corresponding task is deleted from the program.
+    
+#### Change Task Priority
+An existing task can have its priority changed when the user wishes to re-rank the urgency of the task. The task
+ID and the new priority are provided.
+Command executed by user `task /priority -priority HIGH -id 1`
+is passed, the following operations are implemented:
+    * UI receives user input and passes it to Parser class.
+    * Parser checks if the id and priority entered is valid, and executes a corresponding ChangeTaskPriorityCommand 
+    object.
+    * The corresponding task will have its priority updated in the program.
+    
+#### Mark Task as Complete    
+An existing task can be marked as complete when the user completes the task. The task
+ID is provided.
+Command executed by user `task /done -id`
+is passed, the following operations are implemented:
+    * UI receives user input and passes it to Parser class.
+    * Parser checks if the id entered is valid, and executes a corresponding DoneTaskCommand 
+    object.
+    * The corresponding task is marked as done in the program.
+    
+### Sprint
+In SCRUMptious, a Project will be broken down into smaller iterations known as Sprints. The Sprint will contain information about the Tasks allocated for that iteration and Members that are assigned to complete the Tasks.
+
+The following section will explain how the management of Sprints is implemented in the program.
+
+#### Create Sprint
+
+![Figure X: Sequence diagram of CreateSprintCommand](./image/developerguide/createSprint.png "Create Sprint Sequence Diagram")  
```

Comment:
After the CreateSprintCommand command is completed, should the command object return control to the user? And maybe the second optional code block box can be smaller, since the other classes are not running any optional code in that period of time.